src/commands.ts

```diff
@@ -39,33 +66,150 @@ const commands: Record<string, Command> = {
         fn: async (msg) => {
             await msg.channel.send('Use the **/track** command');
         },
+        build() {
+            const builder = new SlashCommandBuilder()
+                .setName('track')
+                .addStringOption(option =>
+                    option
+                        .setName('username')
+                        .setDescription('Username')
+                        .setRequired(true));
+            return buildSlashCommand(builder, this);
```

Comment:
There seems to be a lot of redundancy in these `build()` commands. Is there any way we can do this all in the `buildCommands` loop? Also, we can do `setName` programmatically using inferred info from the command name (ensures consistency), and `setDescription` using the `text` property. The only thing we'd need to define per-command is the list of args (i.e. "options")

Reply:
Also, seems weird to me that we're calling `buildSlashCommand` in the command itself. IMO we should have this be done in `CommandHandler.buildCommands`